lib/rucio/transfertool/fts3_new.py

```diff
@@ -0,0 +1,639 @@
+        :param job_params:   Dictionary containing key/value pairs, for all transfers.
+        :param timeout:      Timeout in seconds.
+        :returns:            FTS transfer identifier.
+        """
+
+        # FTS3 expects 'davs' as the scheme identifier instead of https
+        for file in files:
+            if not file['sources'] or file['sources'] == []:
+                raise Exception('No sources defined')
+
+            new_src_urls = []
+            new_dst_urls = []
+            for url in file['sources']:
+                if url.startswith('https'):
+                    new_src_urls.append(':'.join(['davs'] + url.split(':')[1:]))
+                else:
+                    new_src_urls.append(url)
+            for url in file['destinations']:
+                if url.startswith('https'):
+                    new_dst_urls.append(':'.join(['davs'] + url.split(':')[1:]))
+                else:
+                    new_dst_urls.append(url)
+
+            file['sources'] = new_src_urls
+            file['destinations'] = new_dst_urls
+
+        transfer_id = None
+        expected_transfer_id = None
+        if __USE_DETERMINISTIC_ID:
+            job_params = job_params.copy()
+            job_params["id_generator"] = "deterministic"
+            job_params["sid"] = files[0]['metadata']['request_id']
+            expected_transfer_id = self.__get_deterministic_id(job_params["sid"])
+            logging.debug("Submit bulk transfers in deterministic mode, sid %s, expected transfer id: %s" % (job_params["sid"], expected_transfer_id))
+
+        # bulk submission
+        params_dict = {'files': files, 'params': job_params}
+        params_str = json.dumps(params_dict)
+
+        r = None
+        if self.external_host.startswith('https://'):
+            try:
+                ts = time.time()
+                r = requests.post('%s/jobs' % self.external_host,
+                                  verify=False,
+                                  cert=(__USERCERT, __USERCERT),
+                                  data=params_str,
+                                  headers={'Content-Type': 'application/json'},
+                                  timeout=timeout)
+                record_timer('transfertool.fts3.submit_transfer.%s' % self.__extract_host(self.external_host), (time.time() - ts) * 1000 / len(files))
+            except:
+                logging.warn('Could not submit transfer to %s - %s' % (self.external_host, str(traceback.format_exc())))
+        else:
+            try:
+                ts = time.time()
+                r = requests.post('%s/jobs' % self.external_host,
+                                  data=params_str,
+                                  headers={'Content-Type': 'application/json'},
+                                  timeout=timeout)
+                record_timer('transfertool.fts3.submit_transfer.%s' % self.__extract_host(self.external_host), (time.time() - ts) * 1000 / len(files))
+            except:
```

Comment:
Would be good to catch explicitely the following error : 

SSLError: HTTPSConnectionPool(host='fts.usatlas.bnl.gov', port=8446): Max retries exceeded with url: /jobs (Caused by SSLError(SSLError("bad handshake: SysCallError(104, 'ECONNRESET')",),))